lib/driver.rb

```diff
@@ -0,0 +1,62 @@
+require_relative "csv_record"
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :name, :vin, :trips
+    attr_accessor :status
+
+    def initialize(id:, name:, vin: nil, status: :AVAILABLE, trips: nil)
+      super(id)
+      @name = name
+      if vin.length != 17
+        raise ArgumentError, "vin must be 17 characters long"
+      end
+      @vin = vin
+      @status = status
+      @trips = trips || []
+    end
+
+    def add_trip(trip)
+      @trips << trip
+    end
+
+    def average_rating
+      rating_total = 0
+      if @trips.empty?
+        return 0
+      else
+        @trips.each do |trip|
+          rating_total += trip.rating
+        end
+        return rating_total / @trips.length.to_f.round(2)
```

Comment:
This is a really good opportunity for refactoring to use an Enumerable method, like `sum`, instead of needing to increment `rating_total` in an `each` loop